src/js/controller/controller.js

```diff
@@ -530,7 +523,17 @@ Object.assign(Controller.prototype, {
         }
 
         function _setItem(index) {
-            return _model.setItemIndex(index);
+            _programController.stopVideo();
+
+            const playlist = _model.get('playlist');
+            const length = playlist.length;
+
+            // If looping past the end, or before the beginning
+            index = parseInt(index, 10) || 0;
+            index = (index + length) % length;
```

Comment:
`item = playlist[index]` would return an undefined value for negative values of index > length * 2. eg. index = -7 when length = 3.

Though this shouldn't happen, here's a safe way to handle  either extreme:
```
index %= length;
if (index < 0) {
    index += length;
}
```

Reply:
This was existing logic but I can update it